Source/WebCore/platform/graphics/MIMESniffer.cpp

```diff
@@ -0,0 +1,341 @@
+            //  7. If matched is true, abort these steps and return true.
+            if (sequence[iter] == 0x77 && sequence[iter + 1] == 0x65 && sequence[iter + 2] == 0x62 && sequence[iter + 3] == 0x6d)
+                return true;
+        }
+        //  2. Increment iter by 1.
+        iter++;
+    }
+    //   7. Return false.
+    return false;
+}
+
+static bool matchMP3Header(std::span<const uint8_t> sequence)
+{
+    // To match an mp3 header, using a byte sequence sequence of length length at offset s execute these steps:
+
+    // If length is less than 4, return false.
+    if (sequence.size() < 4)
+        return false;
+
+    // If sequence[s] is not equal to 0xff and sequence[s + 1] & 0xe0 is not equal to 0xe0, return false.
+    if (sequence[0] != 0xff || sequence[1] != 0xe0)
+        return false;
+    // Let layer be the result of sequence[s + 1] & 0x06 >> 1.
+    auto layer = sequence[1] & 0x06 >> 1;
+    // If layer is 0, return false.
+    if (!layer)
+        return false;
+    // Let bit-rate be sequence[s + 2] & 0xf0 >> 4.
+    auto bitRate = sequence[2] & 0xf0 >> 4;
+    // If bit-rate is 15, return false.
+    if (bitRate == 15)
+        return false;
+    // Let sample-rate be sequence[s + 2] & 0x0c >> 2.
+    auto sampleRate = sequence[2] & 0x0c >> 2;
+    // If sample-rate is 3, return false.
+    if (sampleRate == 3)
+        return false;
+    // Let freq be the value given by sample-rate in the table sample-rate.
+    // const uint32_t sampleRateTable[] = { 44100, 48000, 32000 };
+    // auto freq = sampleRateTable[sampleRate];
+    // Let final-layer be the result of 4 - (sequence[s + 1]).
+    auto finalLayer = 4 - sequence[1];
+    // If final-layer & 0x06 >> 1 is not 3, return false.
+    if ((finalLayer & 0x06 >> 1) != 3)
+        return false;
+    // Return true.
+    return true;
+}
+
+struct MP3Frame {
+    uint8_t version;
+    uint32_t bitRate;
+    uint32_t sampleRate;
+    uint8_t pad;
+};
+
+static uint32_t mp3FrameSize(const MP3Frame& frame)
+{
+    // To compute an mp3 frame size, execute these steps:
+    // If version is 1, let scale be 72, else, let scale be 144.
+    auto scale = frame.version == 1 ? 72u : 144u;
+    // Let size be bitrate * scale / freq.
+    uint32_t size = frame.bitRate * scale / frame.sampleRate;
+    // If pad is not zero, increment size by 1.
+    if (frame.pad)
+        size += 1;
+    // Return size.
+    return size;
+}
+
+static MP3Frame parseMP3Frame(std::span<const uint8_t> sequence)
+{
+    // Let version be sequence[s + 1] & 0x18 >> 3.
+    uint8_t version = sequence[1] & 0x18 >> 3;
+    // Let bitrate-index be sequence[s + 2] & 0xf0 >> 4.
+    auto bitrateIndex = sequence[2] & 0xf0 >> 4;
+    // If the version & 0x01 is non-zero, let bitrate be the value given by bitrate-index in the table mp2.5-rates
+    const uint32_t mp25ratesTable[] = { 0, 8000, 16000, 24000, 32000, 40000, 48000, 56000, 64000, 80000, 96000, 112000, 128000, 144000, 160000 };
```

Comment:
Would be nice to use `std::array` in new code. Same comment below.
You can also use `constexpr`.

Reply:
annoying that you have to set the size in the template though, auto-deduction when you want the data to be `const uint8_t` doesn't make for nice syntax